api/src/test/java/org/apache/iceberg/util/RandomUtil.java

```diff
@@ -228,4 +235,54 @@ private static BigInteger randomUnscaled(int precision, Random random) {
 
     return new BigInteger(sb.toString());
   }
+
+  public static List<Object> generateList(
+      Random random, Types.ListType list, Supplier<Object> elementResult) {
```

Comment:
Nit: the supplier names no longer make much sense in this context, but this is minor.